components/browser/state/src/test/java/mozilla/components/browser/state/selector/SelectorsKtTest.kt

```diff
@@ -236,25 +236,94 @@ class SelectorsKtTest {
 
         BrowserState(
             customTabs = listOf(createCustomTab("https://www.mozilla.org", id = "test-id"))
         ).also { state ->
-            assertNull(state.findTabByUrl("https://www.mozilla.com"))
+            assertNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.com", false))
+        }
+    }
+
+    @Test
+    fun `findNormalOrPrivateTabByUrl finds no result looking for a normal tab when only a private tab matches`() {
+        BrowserState(
+            tabs = listOf(createTab("https://www.mozilla.org", id = "test-id", private = true))
+        ).also { state ->
+            assertNotNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true))
+            assertEquals(
+                "https://www.mozilla.org",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true)!!.content.url
+            )
+            assertEquals(
+                "test-id",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true)!!.id
+            )
+        }
+
+        BrowserState(
+            customTabs = listOf(createCustomTab("https://www.mozilla.org", id = "test-id"))
+        ).also { state ->
+            assertNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.com", false))
+        }
+    }
+
+    @Test
+    fun `findNormalOrPrivateTabByUrl finds a private tab with same url or returns null`() {
+        BrowserState(
+            tabs = listOf(createTab("https://www.mozilla.org", id = "test-id", private = true))
+        ).also { state ->
+            assertNotNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true))
+            assertEquals(
+                "https://www.mozilla.org",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true)!!.content.url
+            )
+            assertEquals(
+                "test-id",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true)!!.id
+            )
+        }
+
+        BrowserState(
+            customTabs = listOf(createCustomTab("https://www.mozilla.org", id = "test-id"))
+        ).also { state ->
+            assertNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.com", true))
+        }
+    }
+
+    @Test
+    fun `findNormalOrPrivateTabByUrl finds no result looking for a private tab when only a normal tab matches`() {
+        BrowserState(
+            tabs = listOf(createTab("https://www.mozilla.org", id = "test-id", private = false))
+        ).also { state ->
+            assertNotNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", false))
+            assertEquals(
+                "https://www.mozilla.org",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", false)!!.content.url
+            )
+            assertEquals(
+                "test-id",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", false)!!.id
+            )
+        }
```

Comment:
This doesn't do what the test says ^^. It asserts it can find a normal (private=false) tab by url, but the test says "finds no result looking for a private tab when only a normal tab matches". For that you need to create state with a normal tab and assert that you can't find it when looking for a private tab.